src/transformers/models/opt/modeling_opt.py

```diff
@@ -466,7 +487,34 @@ def forward(
             q = query_states.view(*original_shape)
             k = key_states.view(*original_shape)
             v = value_states.view(*original_shape)
-            attn_output = self.flash_attention_triton(q, k ,v, True, self.scaling).half()
+            # lut: (bsz, self.num_heads, seq_len/BLOCK_M, nnz)
+            # give any lut you want here, but each row should have a valid block
+            nnz = 5
+            lut = torch.tensor([0,1,2,3,4], device=q.device).view(1, nnz)
+            lut = lut.expand(1, 32, 64, nnz).contiguous()
+            attn_output = self.flash_attention_triton(q, k ,v, True, self.scaling, lut, 64, 64).half()
+
+            if not output_attentions:
+                attn_weights_reshaped = None
+
+        elif self._is_flash_attn_triton_decode:
+            original_shape = (bsz, self.num_heads, -1, self.head_dim)
+            q = query_states.view(*original_shape)
+            k = key_states.view(*original_shape)
+            v = value_states.view(*original_shape)
+            # nnz = 3
+            # lut = torch.tensor([i for i in range(nnz)], device=q.device).view(1, nnz)
+            # lut = lut.expand(3, 32, nnz).contiguous()
+            lut = torch.load('test_lut.pt').to(q.device)
```

Comment:
Can you load the lut in init and directly call it here instead of loading it every time